.ci/Jenkinsfile

```diff
@@ -103,6 +108,46 @@ pipeline {
         }
       }
     }
+    stage('Publish Docker Image') {
+      options { skipDefaultCheckout() }
+      when {
+        not { changeRequest() }
+      }
+      environment {
+        DOCKER_IMAGE_SHA_TAG = "${env.GIT_BASE_COMMIT}"
+        DOCKER_IMAGE_LATEST_TAG = "latest"
+        DOCKER_IMAGE_GIT_TAG = "${env.BRANCH_NAME}"
+      }
+      steps {
+        withGithubNotify(context: "Publish Docker images") {
+          cleanup()
+          dir("${BASE_DIR}") {
+            catchError(buildResult: 'UNSTABLE', message: 'Unable to push Docker images', stageResult: 'FAILURE') {
+              dockerLogin(secret: "${env.DOCKER_REGISTRY_SECRET}",
+                registry: "${env.DOCKER_REGISTRY}")
+
+              sh(label: 'Build docker image', script: """
+                # not rebuild the image if it exists in the docker registry
+                if ! docker pull -q ${env.DOCKER_IMAGE}:${env.DOCKER_IMAGE_SHA_TAG} 2> /dev/null ; then
+                  DOCKER_IMAGE="${env.DOCKER_IMAGE}" DOCKER_IMAGE_TAG="${env.DOCKER_IMAGE_SHA_TAG}" make build-docker
+                fi
```

Comment:
If the docker image is already present in the docker registry (docker.elastic.co), it will not build again the docker image.

This should be useful when a git tag has been pushed, since the docker image for that changeset should already be available.